internal/kafka/internal/services/kafka_test.go

```diff
@@ -2601,3 +2734,591 @@ func Test_kafkaService_GetAvailableSizesInRegion(t *testing.T) {
 		})
 	}
 }
+func Test_kafkaService_GetManagedKafkaByClusterID(t *testing.T) {
+	type fields struct {
+		connectionFactory *db.ConnectionFactory
+		keycloakService   sso.KeycloakService
+		kafkaConfig       *config.KafkaConfig
+	}
+	type args struct {
+		clusterID string
+	}
+	kafkaRequestList := dbapi.KafkaList{
+		&dbapi.KafkaRequest{
+			ClusterID: testClusterID,
+			InstanceType: "developer",
+			SizeId: "x2",
+		},
+	}
+	managedkafkaCR, _ := buildManagedKafkaCR(
+		&dbapi.KafkaRequest{
+			ClusterID: testClusterID,
+			InstanceType: "developer",
+			SizeId: "x2",
+		},
+		&config.KafkaConfig{
+			EnableKafkaExternalCertificate: true,
+			SupportedInstanceTypes: &kafkaSupportedInstanceTypesConfig,
+		},
+			&sso.KeycloakServiceMock{
+				GetConfigFunc: func() *keycloak.KeycloakConfig {
+					return &keycloak.KeycloakConfig{
+						EnableAuthenticationOnKafka: true,
+					}
+				},
+				GetRealmConfigFunc: func() *keycloak.KeycloakRealmConfig {
+					return &keycloak.KeycloakRealmConfig{}
+				},
+		})
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []managedkafka.ManagedKafka
+		wantErr *errors.ServiceError
+		setupFn func()
+	}{
+		{
+			name: "should return the kafka by cluster id",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				keycloakService: &sso.KeycloakServiceMock{
+					GetConfigFunc: func() *keycloak.KeycloakConfig {
+						return &keycloak.KeycloakConfig{
+							EnableAuthenticationOnKafka: true,
+						}
+					},
+					GetRealmConfigFunc: func() *keycloak.KeycloakRealmConfig {
+						return &keycloak.KeycloakRealmConfig{}
+					},
+				},
+				kafkaConfig: &config.KafkaConfig{
+					EnableKafkaExternalCertificate: true,
+					SupportedInstanceTypes: &kafkaSupportedInstanceTypesConfig,
+				},
+			},
+			args: args{
+				clusterID: testClusterID,
+			},
+			wantErr: nil,
+			want: []managedkafka.ManagedKafka{*managedkafkaCR},
+			setupFn: func() {
+				mocket.Catcher.Reset()
+				query := fmt.Sprintf(`SELECT * FROM "%s"`, kafkaRequestTableName)
+				response := converters.ConvertKafkaRequestList(kafkaRequestList)
+				mocket.Catcher.NewMock().WithQuery(query).WithReply(response)
+				mocket.Catcher.NewMock().WithExecException().WithQueryException()
+			},
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		tt.setupFn()
+		t.Run(tt.name, func(t *testing.T) {
+			k := &kafkaService{
+				connectionFactory: tt.fields.connectionFactory,
+				keycloakService:   tt.fields.keycloakService,
+				kafkaConfig:       tt.fields.kafkaConfig,
+			}
+			got, got1 := k.GetManagedKafkaByClusterID(tt.args.clusterID)
+			Expect(got).To(Equal(tt.want))
+			Expect(got1).To(Equal(tt.wantErr))
```

Comment:
This is incorrect. Same issue as in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/981#discussion_r862971406.
Additionally please review all the similar checks added in this PR to make sure all of them are fixed.